src/ompl/geometric/planners/rrt/src/STRRTstar.cpp

```diff
@@ -993,7 +1000,10 @@ bool ompl::geometric::STRRTstar::rewireGoalTree(Motion *addedMotion)
 {
     bool solved = false;
     std::vector<Motion *> nbh;
+    tGoal_->setDistanceFunction([this](const Motion *a, const Motion *b) { return distanceFunction(a, b); });
     getNeighbors(tGoal_, addedMotion, nbh);
+    tGoal_->setDistanceFunction([this](const Motion *a, const Motion *b) { return distanceFunction(b, a); });
```

Comment:
4) In ompl::geometric::STRRTstar::rewireGoalTree, when we are finding candidates for rewiring, we expect them to be in the "past", so our query state "addedMotion" is in the future. So, ti satisfy t1<t2, we must change distance function, and pass query state as second argument in distance function. (Inside nearestK and nearestR the query state is passed as second argument "b" )